src/server/protocol.ts

```diff
@@ -2700,6 +2701,42 @@ export interface InlayHintsResponse extends Response {
     body?: InlayHintItem[];
 }
 
+export interface MapCodeRequestArgs {
+    /// The files and changes to try and apply/map.
+    mappings: MapCodeRequestDocumentMapping[];
+
+    /// Edits to apply to the current workspace before performing the mapping.
+    updates?: FileCodeEdits[];
+}
+
+export interface MapCodeRequestDocumentMapping {
+    /// The file for the request (absolute pathname required). Null/undefined
+    /// if specific file is unknown.
+    file?: string;
+
+    /// Optional name of project that contains file
+    projectFileName?: string;
+
+    /// The specific code to map/insert/replace in the file.
+    contents: string[];

```

Comment:
What if you want to remove code? What would `contents` be?

Reply:
there's no specific facility for _removing_ code. If copilot wants to remove code, it simply includes the larger scope, with the code removed from it.

Reply:
I'd like to keep the naming scheme close to what the protocol is, and that's already in place and in use.